src/rainflow.py

```diff
@@ -144,7 +153,44 @@ def count_cycles(series, ndigits=None, left=False, right=False):
     counts = defaultdict(float)
     round_ = _get_round_function(ndigits)
 
-    for low, high, mult in extract_cycles(series, left=left, right=right):
-        delta = round_(abs(high - low))
-        counts[delta] += mult
+    try:
+        max_range = max(series) - min(series)
```

Comment:
This is a breaking change because `count_cycle` will no longer work if `series` is an iterator. That's why you had to change `test_series_with_zero_derivatives`.

Actually we do not really need `max_range` when `binsize` or `ndigits` is given. I suggest that `max_range` is calculated only when `nbins` is given and that an exception is thrown if `series` is an iterator. This can be tested by:

```python
if series is iter(series):
    raise ...
```

Reply:
ok